include/ignition/gazebo/components/JointTransmittedWrench.hh

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+#ifndef IGNITION_GAZEBO_COMPONENTS_JOINTTRANSMITTEDWRENCH_HH_
+#define IGNITION_GAZEBO_COMPONENTS_JOINTTRANSMITTEDWRENCH_HH_
+
+#include <array>
+
+#include <ignition/gazebo/components/Factory.hh>
+#include <ignition/gazebo/components/Component.hh>
+#include <ignition/gazebo/components/Serialization.hh>
+#include <ignition/gazebo/config.hh>
+
+namespace ignition
+{
+namespace gazebo
+{
+// Inline bracket to help doxygen filtering.
+inline namespace IGNITION_GAZEBO_VERSION_NAMESPACE {
+
+/// \brief Data structure that contains torque and force.
+struct Wrench
+{
+  math::Vector3d torque;
+  math::Vector3d force;
```

Comment:
Document all members, this is a good place to mention units and reference frames

Reply:
I feel like this struct should go into a different header so it can be used by more than one component, but I'm not sure where. Would `Types.hh` be a good place?  We do use `msgs::Wrench` for the `ExternalWorldWrenchCmd` component, but I remember a discussion where we said we should avoid using messages as data types for components if possible.

Reply:
You know, I'd lean towards using `ignition::msgs::Wrench` here instead of yet another struct. That also gives us serialization for free. What do you think?

In general, I've been trying to use messages in components as much as possible. This is in relation to https://github.com/ignitionrobotics/ign-gazebo/issues/494

Reply:
Changed to `msgs::Wrench` in https://github.com/ignitionrobotics/ign-gazebo/pull/989/commits/41310933ca384a67559f7b166dbabb0488b617e7. One issue is that the message has a `offset_point` member, which we don't support. I made a note of that in the documentation.